### PR TITLE
Print View - Combined (Basics, Daily, Settings)

### PR DESCRIPTION
**Pull request by cbwebdevelopment**
See https://trello.com/c/zdyHFY0M for full details.

Goes hand-in-hand with the same-named branches in `tideline` and `blip`.

**Most Notable Additions**

- A new Settings print view
  - replaces the current settings print view that was in `blip`
  - required adding `upload` data to the `TidelineData` export in `tideline` to determine latest upload device and active basal schedules.
  - there is no mockup provided in zeplin for this view.  It was modelled after the web view.
  - storybooks have been added to test a variety of devices and basal schedules
- A new Basics print view
  - a significant portion of the basics datamunging logic from `tideline` had to be ported over into `viz`.  In some cases, it's a straight port, updated to ES2015, in other cases, it is changed to a more `viz-like` format (i.e. using the new viz-style `bgBounds` structure for bgTargets.  Where possible, any overlap in logic was updated to use existing `viz` counterparts.
  - it should be noted that although the print views use this new, ported-over and updated logic, the web views still use the logic from tideline.  It would be nice to use the new logic in `viz` for the web view to avoid having to update logic in 2 places, but that falls outside the scope of this card.
  - storybooks have been added to test rendering with different siteChange sources (though, you're local `print-view.json` data has to be from a device compatible with that siteChange source -- i.e. animas, tandem, or medT for cannulaPrime and tubingPrime, omniPod for reservoirChange)

**Other notable changes**
- Daily print view updates/additions
  - added scheduled basal rate values on top of basal rendering
  - added dividers between daily charts
  - added dashed lines to denote the upper and lower bg targets in the bg rendering charts
- General copy updates
  - updated footer text on all pages to include 'Questions or feedback? Please email support@tidepool.org or visit support.tidepool.org.'
  - changed 'Printed from Tidepool on:' in header to 'Printed On:'
  - added 'DOB: ' before patient's birthdate in header
- Added a 'Combined' print view storybook item to print all 3 views together.  Please see the `blip` PR for instructions on generating the local data for the new print views (https://github.com/tidepool-org/blip/pull/450)
- FIxed the issue where not having the local data files (which are now renamed, and in a single file) would prevent the storybook from rendering

### CODE REVIEW

**Comment by cbwebdevelopment** [comment]
[<L> [blip] Print View - Combined (Basics, Daily, Settings)](https://trello.com/c/zdyHFY0M/154-l-blip-print-view-combined-basics-daily-settings)

File: data/guid.js [hunk 1]
```diff
@@ -15,5 +15,11 @@
  * == BSD2 LICENSE ==
  */
 
-export const DISPLAY_VIEW = 'display';
-export const PRINT_VIEW = 'print';
+export default function createGUID() {
```
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply data/guid.js]: I'm sure there is.  I can look.  This is currently only used for creating test data, and is just copied over from tideline for the sake of porting over some of the data munging tests.

Either way, I should have this in the `types.js` file, and not as it's own export, so it's more clear that it's not intended for use in application code.

File: data/types.js [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+ * not, you can obtain one from Tidepool Project at tidepool.org.
+ * == BSD2 LICENSE ==
+ */
+
+import _ from 'lodash';
+
+import guid from './guid';
+import { addDuration } from '../src/utils/datetime';
+
+// constants
```
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply data/types.js]: Agreed.  I tried to clean up the `tideline` code a little as I ported stuff over, but there's probably a few things I missed like this.

File: data/guid.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (jh-bate) [reply data/guid.js]: yeah cool - I just saw `data/guid` and thought "really" ... but yeah as a generator for test data that seems ok

**Comment by cbwebdevelopment** [comment]
@jh-bate I’ve made a few fixes for the print views.  Just pull blip and reinstall to test.
Fixes are:
- Setting `medtronic` as the manufacturer when `carelink` is the source for the settings view.
- Fix for basal rate rendering errors (this is what was causing the infinite hang — due to some miscalculations from accidentally including non-scheduled rates in some of the calculations, it was trying to render the text off of the printable area of the pdf)
- Small font style fix for when a table spills over onto the next page during rendering (one of the rows was getting rendered with the smaller footer font style)
- Fix to not show the 'days since' text in the site changes calendar for the first sitechange event.  Currently, this is still broken in the web view (shows `NaN days`, which is what the print view was showing before the fix)

File: data/guid.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply data/guid.js]: Fixed.

File: data/types.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbwebdevelopment) [reply data/types.js]: Fixed.

**Pull request merged**